src/button-group/__tests__/button-group-children.test.tsx

```diff
@@ -0,0 +1,29 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import Button from '../../../lib/components/button';
+import ButtonGroup from '../../../lib/components/button-group';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+describe('ButtonGroup children property', () => {
+  test('renders children', () => {
```

Comment:
I would add a couple more tests:
* Rendering a button w/o testid (assert warning, make sure it won't crash);
* Re-rendering the button group with an extra button (make sure it is included to the keyboard nav);
* Re-rendering the button group with focused button gone (make sure re-focusing occurs).